File: src/ClickUpClient/Models/TaskSummary.cs
namespace ClickUpClient.Models;

/// <summary>
/// エージェント向けの整形済みタスク DTO。
/// TaskSummary 自体がツリーノードを兼ねており、Subtasks に子タスクをネストで保持する。
/// </summary>
public sealed record TaskSummary(
    string Id,
    string Name,
    /// <summary>ステータス表示名 (例: "in progress", "to do")</summary>
    string Status,
    /// <summary>優先度表示名 (例: "urgent", "normal")。未設定の場合は null</summary>
    string? Priority,
    /// <summary>親タスクの ID。null = ルートタスク</summary>
    string? ParentId,
    string Url,
    DateTimeOffset? DueDate,
    string? Description,
    string ListId,
    string ListName,
    DateTimeOffset CreatedAt,
    DateTimeOffset UpdatedAt,
    IReadOnlyList<TaskSummary> Subtasks
);

File: internal/models/task_summary.go
// internal/models/task_summary.go
package models

import "time"

// TaskSummary はエージェント向けの整形済みタスク DTO。
// Subtasks に子タスクをネストして保持するツリーノードを兼ねる。
type TaskSummary struct {
	ID          string        `json:"id"`
	Name        string        `json:"name"`
	Status      string        `json:"status"`
	Priority    *string       `json:"priority"`
	ParentID    *string       `json:"parentId"`
	URL         string        `json:"url"`
	StartDate   *time.Time    `json:"startDate,omitempty"`
	DueDate     *time.Time    `json:"dueDate"`
	Description *string       `json:"description"`
	ListID      string        `json:"listId"`
	ListName    string        `json:"listName"`
	CreatedAt   time.Time     `json:"createdAt"`
	UpdatedAt   time.Time     `json:"updatedAt"`
	Subtasks    []TaskSummary `json:"subtasks"`
}
